### ISSUE TITLE
service zfs-zed won't load

### ISSUE THREAD

**Issue by ochbob** [zfsonlinux/zfs#5553]
Hi all,

i finished my switch from archive.zfsonlinux.org  0.6.5.7 repo to 0.6.5.8 debian-backports.

Everythings seems to be ok, after reboot pool just mount fine.
But one service from ZFS does not load:

```
service zfs-zed status
● zfs-zed.service - ZFS Event Daemon (zed)
   Loaded: loaded (/lib/systemd/system/zfs-zed.service; enabled)
   Active: failed (Result: exit-code) since lun. 2017-01-02 20:46:12 CET; 8min ago
     Docs: man:zed(8)
  Process: 9101 ExecStart=/sbin/zed -F (code=exited, status=203/EXEC)
 Main PID: 9101 (code=exited, status=203/EXEC)

janv. 02 20:46:12 LINOLOGY systemd[1]: Started ZFS Event Daemon (zed).
janv. 02 20:46:12 LINOLOGY systemd[1]: zfs-zed.service: main process exited, code=exited, status=203/EXEC
janv. 02 20:46:12 LINOLOGY systemd[1]: Unit zfs-zed.service entered failed state.
```

I tried to uninstall and reinstall zfs-zed from backport, but still the same when i try to start service:

`systemctl start zfs-zed.service`

My configuration:

Debian Jessie 8.2
4.8.15 kernel

```
> dpkg -l | grep zfs
> ii  libzfs2linux                                                     0.6.5.8-2~bpo8+1                     amd64        OpenZFS filesystem library for Linux
> ii  zfs-dkms                                                         0.6.5.8-2~bpo8+1                     all          OpenZFS filesystem kernel modules for Linux
> ii  zfs-zed                                                          0.6.5.8-2~bpo8+1                     amd64        OpenZFS Event Daemon
> ii  zfsutils-linux                                                   0.6.5.8-2~bpo8+1                     amd64        command-line tools to manage OpenZFS filesystems
```

Any recommendation ?

thanks !

**Comment by ochbob** [comment]
Ok, found the solution.

i look at "/sbin/" folder, but didn't see "zed" file which call by the zfs-zed.service (/lib/systemd/system/)

```
[Unit]
Description=ZFS Event Daemon (zed)
Documentation=man:zed(8)
After=zfs-import-cache.service
After=zfs-import-scan.service

[Service]
ExecStart=/sbin/zed -F
Restart=on-abort

[Install]
Alias=zed.service
WantedBy=zfs.target
```

So i search and found the "zed" files which call by the zfs-zed.service into "/usr/sbin/".

I copy the zed files into /sbin/ and everything works fine now when i start the service.

kind of "bug" or something else ?

**Comment by loli10K** [comment]
@ochbob  this seems more like a debian-specific packaging issue (https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=849813) rather than a ZoL bug ...

```
root@debian-8-zfs:~# mktemp -d
/tmp/tmp.u3PvD2CAW4
root@debian-8-zfs:~# curl -s http://ftp.debian.org/debian/pool/contrib/z/zfs-linux/zfs-zed_0.6.5.8-2~bpo8+1_amd64.deb -o package.deb
root@debian-8-zfs:~# dpkg -x package.deb /tmp/tmp.u3PvD2CAW4
root@debian-8-zfs:~# grep ExecStart= /tmp/tmp.u3PvD2CAW4/lib/systemd/system/zfs-zed.service 
ExecStart=/sbin/zed -F
root@debian-8-zfs:~# find /tmp/tmp.u3PvD2CAW4 -name zed
/tmp/tmp.u3PvD2CAW4/usr/sbin/zed
root@debian-8-zfs:~# 
```

`@sbindir@` is correctly used in https://github.com/zfsonlinux/zfs/blob/zfs-0.6.5.8/etc/systemd/system/zfs-zed.service.in#L8, but the executable `zed` is installed in `/usr/sbin` ... seems like that .deb archive was packaged from two differenlty `./configure`'d ZoL builds.

**Comment by loli10K** [comment]
@ochbob Fixed in 0.6.5.8-3: https://anonscm.debian.org/cgit/pkg-zfsonlinux/zfs.git/commit/?h=debian/0.6.5.8-3&id=f39f1f07fbb93efacd89555b3e7dc80753b52ffc
